### PR TITLE
Add is-triangle exercise + solution

### PR DESCRIPTION
**Pull request by vaibhavarora14**
This solution is based on `Triangle inequality theorem`, which states
sum of any 2 sides of triangle, should be greater than 3rd one.

triangle-inequality-theorem

### CODE REVIEW

File: js-exercises/is-triangle/isTriangle.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  return result;
+}
+
+function isPositiveInteger(...args) {
+  let result = true;
+
+  for (const arg of args) {
+    if (arg !== Math.abs(arg)) {
```
Comment (ArfatSalman) [inline js-exercises/is-triangle/isTriangle.js:18]: Since the function is `isPositiveInteger`, won't the `args` be numbers?

Comment (ArfatSalman) [reply js-exercises/is-triangle/isTriangle.js:18]: Also, why not this?
```suggestion
    if (arg < 0) {
```

File: js-exercises/is-triangle/isTriangle.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  }
+
+  return result;
+}
+
+function isPositiveInteger(...args) {
+  let result = true;
+
+  for (const arg of args) {
```
Comment (ArfatSalman) [inline js-exercises/is-triangle/isTriangle.js:17]: Would this work?
```suggestion
args.every(arg => arg > 0)
```

File: js-exercises/is-triangle/isTriangle.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+function isNumber(...args) {
+  let result = true;
+
+  for (const arg of args) {
```
Comment (ArfatSalman) [inline js-exercises/is-triangle/isTriangle.js:4]:
```suggestion
args.every(arg => typeof === 'number')
```

File: js-exercises/is-triangle/isTriangle.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  return result;
+}
+
+function isPositiveInteger(...args) {
+  let result = true;
+
+  for (const arg of args) {
+    if (arg !== Math.abs(arg)) {
+      result = false;
+      break;
```
Comment (ArfatSalman) [inline js-exercises/is-triangle/isTriangle.js:20]: `break` is not a good idea at all.

File: js-exercises/is-triangle/isTriangle.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  }
+
+  return result;
+}
+
+function isPositiveInteger(...args) {
+  let result = true;
```
Comment (ArfatSalman) [inline js-exercises/is-triangle/isTriangle.js:15]: Not required when you use array functions.

**Review by ArfatSalman — CHANGES_REQUESTED** [review]

File: js-exercises/is-triangle/isTriangle.js (hunk 2 above)
Comment (varora1406) [reply js-exercises/is-triangle/isTriangle.js]: Yes, changed in latest commit with `every`

File: js-exercises/is-triangle/isTriangle.js (hunk 4 above)
Comment (varora1406) [reply js-exercises/is-triangle/isTriangle.js]: As we found alternate, so this is not longer used now

File: js-exercises/is-triangle/isTriangle.js (hunk 5 above)
Comment (varora1406) [reply js-exercises/is-triangle/isTriangle.js]: correct, no longer exist in latest code pushed

File: js-exercises/is-triangle/isTriangle.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
+function isNumber(...args) {
+  let result = true;
+
+  for (const arg of args) {
+    if (typeof arg !== 'number') {
+      result = false;
```
[the remark this answers is not in the document]
Comment (varora1406) [reply js-exercises/is-triangle/isTriangle.js]: Right, have a look at latest code. This block has changed and sort of removed.

File: js-exercises/is-triangle/isTriangle.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (varora1406) [reply js-exercises/is-triangle/isTriangle.js]: yes, it's not in latest code now.
Have a look at it. 🙂

File: js-exercises/is-triangle/isTriangle.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+  if (side1 + side2 <= side3) {
+    return false;
+  }
+
+  if (side2 + side3 <= side1) {
+    return false;
+  }
+
+  return (side3 + side1 <= side2);
```
Comment (ahhda) [inline js-exercises/is-triangle/isTriangle.js:20]: This kind of breaks my mental model and makes it look like you are giving some kind of preference to some sides. Again not a required change, but something like this makes it look better

```js
const sumofAB = sideA + sideB;
const sumofBC = sideB + sideC;
const sumofCA = sideC + sideA;
return (sumofAB > sideC && sumofBC > sideA && sumofCA > sideB)
```

File: js-exercises/is-triangle/isTriangle.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+const isNumber = (arg) => typeof arg === 'number';
+
+const isPositiveNumber = (arg) => isNumber(arg) && (arg > 0);
+
+const arePositiveNumbers = (args) => args.every(isPositiveNumber);
```
Comment (ahhda) [inline js-exercises/is-triangle/isTriangle.js:5]: Great use of predicates! 👍

File: js-exercises/.eslintrc [hunk 9]
```diff
@@ -6,6 +6,7 @@
   "rules": {
     "no-restricted-syntax": 0,
     "arrow-parens": 0,
-    "import/prefer-default-export": 0
+    "import/prefer-default-export": 0,
+    "linebreak-style": 0
```
Comment (ahhda) [inline js-exercises/.eslintrc:10]: As mentioned in the guidelines before, if you are making changes to ESlint or something which is directly not related to the question, please create a separate PR for that.

**Review by ahhda — CHANGES_REQUESTED** [review]
Good work! Just a reminder, separate PR's for different kinds of changes.

File: js-exercises/.eslintrc (hunk 9 above)
Comment (varora1406) [reply js-exercises/.eslintrc:10]: right, will take care further

File: js-exercises/is-triangle/isTriangle.js (hunk 7 above)
Comment (varora1406) [reply js-exercises/is-triangle/isTriangle.js]: done 6e34cd1bb192b07eded55b418f0012cafcbc00ff
Updating subject of PR as well soon

**Comment by vaibhavarora14** [comment]
> 
> 
> Good work! Just a reminder, separate PR's for different kinds of changes.

Will take care from next time 👍

**Pull request closed without merging**